miio/dreamevacuum.py

```diff
@@ -0,0 +1,329 @@
+    # : str = field(metadata={'siid': 25, 'piid': 1, 'access': ['read', 'notify']})
+}
+
+
+class ChargingState(Enum):
+    Charging = 1
+    Not_charging = 2
+    Charging2 = 4
+    Go_charging = 5
+
+
+class CleaningMode(Enum):
+    Quiet = 0
+    Default = 1
+    Medium = 2
+    Strong = 3
+
+
+# TODO: Replace with correct operating modes
+class OperatingMode(Enum):
+    FIXME1 = 0
+    FIXME2 = 17
+    FIXME3 = 1
+    FIXME4 = 14
+
+
+class FaultStatus(Enum):
+    No_faults = 0
+
+
+class DeviceStatus(Enum):
+    Sweeping = 1
+    Idle = 2
+    Paused = 3
+    Error = 4
+    Go_charging = 5
+    Charging = 6
+
+
+class DreameVacuumStatus:
+    def __init__(self, data):
+        self.data = data
+
+    @property
+    def battery_level(self) -> str:
+        return self.data["battery_level"]
+
+    @property
+    def brush_left_time(self) -> str:
+        return self.data["brush_left_time"]
+
+    @property
+    def brush_left_time2(self) -> str:
+        return self.data["brush_left_time2"]
+
+    @property
+    def brush_life_level2(self) -> str:
+        return self.data["brush_life_level2"]
+
+    @property
+    def brush_life_level(self) -> str:
+        return self.data["brush_life_level"]
+
+    @property
+    def filter_left_time(self) -> str:
+        return self.data["filter_left_time"]
+
+    @property
+    def filter_life_level(self) -> str:
+        return self.data["filter_life_level"]
+
+    @property
+    def device_fault(self) -> FaultStatus:
+        return FaultStatus(self.data["device_fault"])
+
+    @property
+    def charging_state(self) -> ChargingState:
+        return ChargingState(self.data["charging_state"])
+
+    @property
+    def operating_mode(self) -> OperatingMode:
+        return OperatingMode(self.data["operating_mode"])
+
+    @property
+    def cleaning_mode(self) -> CleaningMode:
+        return CleaningMode(self.data["cleaning_mode"])
+
+    @property
+    def area(self) -> str:
+        return self.data["area"]
+
+    @property
+    def device_status(self) -> DeviceStatus:
+        return DeviceStatus(self.data["device_status"])
+
+    @property
+    def life_sieve(self) -> str:
+        return self.data["life_sieve"]
+
+    @property
+    def life_brush_side(self) -> str:
+        return self.data["life_brush_side"]
+
+    @property
+    def life_brush_main(self) -> str:
+        return self.data["life_brush_main"]
+
+    @property
+    def timer_enable(self) -> str:
+        return self.data["timer_enable"]
+
+    @property
+    def start_time(self) -> str:
+        return self.data["start_time"]
+
+    @property
+    def stop_time(self) -> str:
+        return self.data["stop_time"]
+
+    @property
+    def map_view(self) -> str:
+        return self.data["map_view"]
+
+    @property
+    def volume(self) -> str:
+        return self.data["volume"]
+
+    @property
+    def voice_package(self) -> str:
+        return self.data["voice_package"]
+
+
+class DreameVacuum(MiotDevice):
```

Comment:
Is it the correct way to extend from MiotDevice or should i directly extend from Device?